java/src/main/scala/prometheus4cats/javasimpleclient/JavaMetricRegistry.scala

```diff
@@ -63,8 +66,51 @@ class JavaMetricRegistry[F[_]: Async: Logger] private (
     case _ => name.show
   }
 
-  private def timeoutCallback[A](fa: F[A], onError: A, stringName: String): A =
-    Utils.timeoutCallback(dispatcher, callbackTimeout, fa, onError, s"Affected metric: '$stringName'")
+  private def trackErrors[A](state: Ref[F, Set[String]], stringName: String, onContains: F[A], onContainsNot: F[A]) =
+    state.modify { current =>
+      if (current.contains(stringName)) (current, onContains) else (current + stringName, onContainsNot)
+    }.flatten
+
+  private def incErrorCounter(stringName: String, reason: String) =
+    Sync[F].delay(callbackErrorCounter.labels(stringName, reason).inc())
+
+  private def timeoutCallback[A](fa: F[A], empty: A, stringName: String): A = {
+    lazy val incTimeout = incErrorCounter(stringName, "timeout").as(empty)
```

Comment:
Interesting nitpick: this is only called once so you could also make it a `def` to avoid allocating if not required. Not sure how big the synchronization overhead of lazy is? Almost certainly it literally doesn't matter, just curious! 😂